checker/cost.go

```diff
@@ -599,3 +605,30 @@ func (c *coster) newAstNode(e *exprpb.Expr) *astNode {
 	}
 	return &astNode{path: path, t: c.getType(e), expr: e, derivedSize: derivedSize}
 }
+
+func isScalar(t *exprpb.Type) bool {
+	switch kindOf(t) {
+	case kindPrimitive:
+		if t.GetPrimitive() != exprpb.Type_STRING && t.GetPrimitive() != exprpb.Type_BYTES {
+			return true
+		}
+	case kindWellKnown:
+		if t.GetWellKnown() == exprpb.Type_DURATION || t.GetWellKnown() == exprpb.Type_TIMESTAMP {
+			return true
+		}
+	case kindObject:
+		switch t.GetMessageType() {
+		case "google.protobuf.Duration", "google.protobuf.Timestamp",
+			"google.protobuf.BoolValue", "google.protobuf.BytesValue",
+			"google.protobuf.DoubleValue", "google.protobuf.FloatValue",
+			"google.protobuf.Int32Value", "google.protobuf.Int64Value",
+			"google.protobuf.UInt32Value", "google.protobuf.UInt64Value":
```

Comment:
Do we have the test case for when this can happen?

Reply:
(marking this as resolved due to [this comment](https://github.com/google/cel-go/pull/571#discussion_r946080502))